src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -691,23 +695,22 @@ internal static bool Dispatch()
                     //
                     // Execute the workitem outside of any finally blocks, so that it can be aborted if needed.
                     //
+#if FEATURE_OBJCWRAPPERS
+                    if (ThreadPool.EnableDispatchAutoReleasePool)
+                    {
+                        Debug.Assert(OperatingSystem.IsMacOS());
+                        DispatchItemWithAutoReleasePool(workItem, currentThread);
+                    }
+                    else
+#endif
```

Comment:
I don't think this should be under an ifdef (or at least, not under `FEATURE_OBJCWRAPPERS`

We should use a linker substitution to choose whether we're going to use the AutoReleasePool dispatch or the plain one.